docs/docs/jsx-in-depth.md

```diff
@@ -0,0 +1,300 @@
+
+### Props Default to "True"
+
+If you pass no value for a prop, it defaults to `true`. These two JSX expressions are equivalent:
+
+```js
+<MyTextBox autocomplete />
+
+<MyTextBox autocomplete={true} />
+```
+
+In general, we don't recommend using this. It's more consistent to just use the second form and explicitly pass `true`. This behavior is just there so that it matches the behavior of HTML.
+
+### Spread Attributes
+
+If you already have `props` as an object, and you want to pass it in JSX, you can use `...` as a "spread" operator to pass the whole props object. These two render functions are equivalent:
+
+```js
+function render1() {
+  var props = {left: 'ben', right: 'hector'};
+  return <MyComponent {...props} />;
+}
+
+function render2() {
+  return <MyComponent left="ben" right="hector" />;
+}
+```
+
+Spread attributes can be useful when you are building generic containers. However, they can also make your code messy by making it easy to pass a lot of irrelevant props to components that don't care about them. We recommend that you use this syntax sparingly.
+
+## Children in JSX
+
+In JSX expressions that contain both an opening tag and a closing tag, the content between those tags is passed as a special prop: `props.children`. There are several different ways to pass children:
+
+### String Literals
+
+You can put a string between the opening and closing tags and `props.children` will just be that string. This is useful for many of the built-in HTML elements. For example:
+
+```js
+<MyComponent>Hello world!</MyComponent>
+```
+
+This is valid JSX, and `props.children` in `MyComponent` will simply be the string `"Hello world!"`. HTML is unescaped, so you can generally write JSX just like you would write HTML in this way:
+
+```html
+<div>This is valid HTML &amp; JSX at the same time.</div>
+```
+
+### JSX Children
+
+You can provide more JSX as the children. This is useful for displaying nested components:
+
+```js
+<MyContainer>
+  <MyFirstComponent />
+  <MySecondComponent />
+</MyContainer>
+```
+
+You can mix together different types of children, so you can use string literals together with JSX children. This is another way in which JSX is like HTML, so that this is both valid JSX and valid HTML:
+
+```html
+<div>
+  Here is a list:
+  <ul>
+    <li>Item 1</li>
+    <li>Item 2</li>
+  </ul>
+</div>
+```
+
+A React component can't return multiple JSX elements, but a single JSX element can have multiple children, so if you want a component to render multiple things you can wrap it in a `div` like this.
+
+When you pass JSX as the children, `props.children` does not actually let you access the child component from the parent. Instead, it lets you access the return value of `React.createElement` calls using the `React.Children` helper library. Generally, accessing children in this way is only useful to return in a render function, rather than modifying or altering the children.
```

Comment:
I'm just going to delete it - it probably belongs more appropriately in somewhere that discusses "components vs instances vs elements" rather than "all about jsx" because it's not really jsx-specific